pandas/core/groupby/generic.py

```diff
@@ -410,7 +410,9 @@ def first_not_none(values):
                         if (isinstance(v.index, MultiIndex) or
                                 key_index is None or
                                 isinstance(key_index, MultiIndex)):
-                            stacked_values = np.vstack(map(np.asarray, values))
+                            stacked_values = np.vstack([
```

Comment:
In case you didn't follow the issue, the original failure came from us passing generator expressions to `vstack`, and the new `__array_function__` stuff in NumPy dev exhausted those generators. So we just build a list instead (which NumPy was doing anyway)

Reply:
yep saw that